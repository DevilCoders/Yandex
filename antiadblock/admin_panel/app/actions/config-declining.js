export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

