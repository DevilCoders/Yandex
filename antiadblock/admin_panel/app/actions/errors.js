export const SET_ERRORS = 'SET_ERRORS';
export const CLEAR_ERRORS = 'CLEAR_ERRORS';

export function setGlobalErrors(errors) {
    return setErrors('global', errors);
}

export function clearGlobalErrors() {
    return clearErrors('global');
}

export function setErrors(id, errors) {
    return {
        type: SET_ERRORS,
        id,
        errors
    };
}

export function clearErrors(id) {
    return {
        type: CLEAR_ERRORS,
        id
    };
}
