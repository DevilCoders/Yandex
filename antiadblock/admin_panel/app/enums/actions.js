export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
