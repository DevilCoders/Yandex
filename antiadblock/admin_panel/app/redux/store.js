import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
