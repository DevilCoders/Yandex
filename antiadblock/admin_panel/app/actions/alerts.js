import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
