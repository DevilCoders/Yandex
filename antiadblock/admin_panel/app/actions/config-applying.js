import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
