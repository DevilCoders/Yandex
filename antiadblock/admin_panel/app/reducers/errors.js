import * as errorsActions from 'app/actions/errors';
import update from 'app/lib/update';

export default function(state, action) {
    const {type} = action;

    state = state || {
        byId: {}
    };

    switch (type) {
        case errorsActions.SET_ERRORS: {
            return update(state, {
                byId: {
                    $merge: {
                        [action.id]: action.errors
                    }
                }
            });
        }
        case errorsActions.CLEAR_ERRORS: {
            return update(state, {
                byId: {
                    $merge: {
                        [action.id]: []
                    }
                }
            });
        }
        default: {
            return state;
        }
    }
}

export function getGlobal(state) {
    const global = state.byId.global || [];
    return global;
}

export function getDashboard(state) {
    const dashboard = state.byId.dashboard || [];
    return dashboard;
}

export function getById(state, id) {
    return state.byId[id];
}
