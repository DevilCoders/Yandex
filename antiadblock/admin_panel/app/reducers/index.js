import {combineReducers} from 'redux';

import alerts from './alerts';
import service from './service';
import services from './services';
import configPreview from './config-preview';
import configApplying from './config-applying';
import configDiffModal from './config-diff-modal';
import ticketCreation from './ticket-creation';
import confirmDialog from './confirm-dialog';
import configDeclining from './config-declining';
import createLabel from './create-label';
import experimentApplying from './experiment-applying';
import errors from './errors';
import metric from './metric';
import search from './search';
import heatmap from './heatmap';
import user from './user';
import profileDiffModal from './profile-diff-modal';

import config from './config/config';

const appReducer = combineReducers({
    service,
    services,
    configPreview,
    configApplying,
    configDiffModal,
    confirmDialog,
    configDeclining,
    createLabel,
    experimentApplying,
    metric,
    errors,
    search,
    heatmap,
    alerts,
    user,
    config,
    profileDiffModal,
    ticketCreation
});

export default appReducer;

export function getService(state) {
    return state.service;
}

export function getServices(state) {
    return state.services;
}

export function getConfigPreview(state) {
    return state.configPreview;
}

export function getConfigApplying(state) {
    return state.configApplying;
}

export function getConfigDiffModal(state) {
    return state.configDiffModal;
}

export function getConfigDeclining(state) {
    return state.configDeclining;
}

export function getExperimentApplying(state) {
    return state.experimentApplying;
}

export function getTicketCreation(state) {
    return state.ticketCreation;
}

export function getMetric(state, serviceId) {
    return state.metric[serviceId];
}

export function getErrors(state) {
    return state.errors;
}

export function getSearch(state) {
    return state.search;
}

export function getHeatmap(state) {
    return state.heatmap;
}

export function getSearchById(state, id) {
    return state.search.byId[id];
}

export function getAlerts(state) {
    return state.alerts;
}

export function getUser(state) {
    return state.user;
}

export function getConfig(state) {
    return state.config;
}

export function getProfileDiffModal(state) {
    return state.profileDiffModal;
}

export function getCreateLabel(state) {
    return state.createLabel;
}
