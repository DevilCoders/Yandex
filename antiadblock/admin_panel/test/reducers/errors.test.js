import reducer from 'app/reducers/errors';
import {getGlobal, getById} from 'app/reducers/errors';
import * as errorsActionFixtures from '../fixtures/actions/errors';
import {ERRORS} from '../fixtures/store';

describe('errors', () => {
    const initialState = ERRORS;

    describe('reducer', () => {
        it('should return the initial state', () => {
            expect(reducer(undefined, {})).toMatchSnapshot();
        });

        it('should handle SET_ERRORS', () => {
            expect(reducer(initialState, errorsActionFixtures.SET_ERRORS)).toMatchSnapshot();
        });

        it('should handle CLEAR_ERRORS', () => {
            expect(reducer(initialState, errorsActionFixtures.CLEAR_ERRORS)).toMatchSnapshot();
        });
    });

    describe('selectors', () => {
        it('getGlobal should return array with id global', () => {
            expect(getGlobal(initialState)).toMatchSnapshot();
        });

        it('getById should return array with id1', () => {
            expect(getById(initialState, 'id1')).toMatchSnapshot();
        });
    });
});
