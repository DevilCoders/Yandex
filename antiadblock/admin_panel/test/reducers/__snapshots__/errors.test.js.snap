// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors reducer should handle CLEAR_ERRORS 1`] = `
Object {
  "byId": Object {
    "global": Array [],
    "id1": Array [
      "message 1",
      "message 2",
    ],
    "id2": Array [
      "message 3",
      "message 4",
    ],
  },
}
`;

exports[`errors reducer should handle SET_ERRORS 1`] = `
Object {
  "byId": Object {
    "global": Array [
      "error message 1",
      "error message 2",
    ],
    "id1": Array [
      "message 1",
      "message 2",
    ],
    "id2": Array [
      "message 3",
      "message 4",
    ],
  },
}
`;

exports[`errors reducer should return the initial state 1`] = `
Object {
  "byId": Object {},
}
`;

exports[`errors selectors getById should return array with id1 1`] = `
Array [
  "message 1",
  "message 2",
]
`;

exports[`errors selectors getGlobal should return array with id global 1`] = `
Array [
  "global message",
]
`;
